boards/shields/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+foreach(shield_dir ${SHIELD_DIRS})
```

Comment:
A shield folder may contain the definition of multiple shields, for example: https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/shields/x_nucleo_iks02a1

Although the likelihood of two or more shields from same shield directory being applied simultaneously is low, then we better prepare for such case, meaning that all duplicates in `SHIELD_DIRS` should be properly handled to avoid including the same CMakeLists.txt file multiple times.